apps/api/src/app/workflows-v2/workflow.controller.e2e.ts

```diff
@@ -519,6 +521,26 @@ describe('Workflow Controller E2E API Testing', () => {
         },
       });
     });
+    it('should serve previous step variables with payload schema', async () => {
+      const steps = [
+        buildDigestStep(),
+        { ...buildInAppStep(), controlValues: { subject: 'Welcome to our newsletter {{payload.inAppSubjectText}}' } },
+      ];
+      const createWorkflowDto: CreateWorkflowDto = buildCreateWorkflowDto('', { steps });
+      const res = await workflowsClient.createWorkflow(createWorkflowDto);
+      if (!res.isSuccessResult()) {
+        throw new Error(res.error!.responseText);
+      }
+      const novuRestResult = await workflowsClient.getWorkflowStepData(res.value._id, res.value.steps[1]._id);
+      if (!novuRestResult.isSuccessResult()) {
+        throw new Error(novuRestResult.error!.responseText);
+      }
+      const { variables } = novuRestResult.value;
+      const variableList = getJsonSchemaPrimitiveProperties(variables as JSONSchemaDto);
+      console.log(variableList);
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common process. Try to leave a link to the process description so that the reader knows how to resolve the chore.  -->
☑ **chore:** remove logs‏